test/embedded/once-multithreaded.swift

```diff
@@ -0,0 +1,66 @@
+// RUN: %empty-directory(%t)
+// RUN: %{python} %utils/split_file.py -o %t %s
+
+// RUN: %target-swift-frontend %t/Main.swift %S/Inputs/print.swift -import-bridging-header %t/BridgingHeader.h -Rmodule-loading -enable-experimental-feature Embedded -c -o %t/main.o
+// RUN: %target-clang %t/main.o -o %t/a.out -dead_strip
+// RUN: %target-run %t/a.out | %FileCheck %s
+
+// REQUIRES: executable_test
+// REQUIRES: optimized_stdlib
+// REQUIRES: VENDOR=apple
+// REQUIRES: OS=macosx
+
+// BEGIN BridgingHeader.h
+
+typedef void *pthread_t;
+int pthread_create(pthread_t *thread, const void *attr, void *(*start_routine)(void *), void *arg);
+int pthread_join(pthread_t thread, void **value_ptr);
+unsigned int sleep(unsigned int);
+int usleep(unsigned int);
+
+// BEGIN Main.swift
+
+public struct MyStructWithAnExpensiveInitializer {
+  static var singleton = MyStructWithAnExpensiveInitializer()
+  static var n = 0
+
+  init() {
+    print("MyStructWithAnExpensiveInitializer.init")
```

Comment:
it might be good to have a recursive initializer test, or dependent ones.

Reply:
Added a test for dependent initializers. Having recursion / a cycle in dependent initializers is not valid Swift, correct?

Reply:
Right:

```
<stdin>:1:1: error: circular reference
let a = a
```

If you manage to hide it from the compiler, I believe you get a runtime error.